### ISSUE TITLE
Add support for writing JSON

### ISSUE THREAD

**Issue by snorfalorpagus** [pywr/pywr#322]
Add support for create a JSON representation of a `Model` from an existing model.

e.g.

```
data = model.dumps()
```

**Comment by snorfalorpagus** [comment]
Implementing this should highlight where parameters don't allow introspection, i.e. once created their information is trapped in cython-space.

**Comment by jetuk** [comment]
This will require a write method on each object?

**Comment by snorfalorpagus** [comment]
Yes. Any object that currently has a load method will require a dump method.

**Comment by jetuk** [comment]
It might be possible to use marshmallow: https://marshmallow.readthedocs.io/en/latest/

Not sure how fiddly this will get because of how pywr actually loads the different objects (i.e. non-linear). How that fits with the `@post_load` decorate I'm not sure.
